server/src/controllers/client_timeline_entries/addClientTimelineEntry.js

```diff
@@ -0,0 +1,96 @@
+const logger = require("pino")();
+const ClientTimelineEntry = require("../../models/client_timeline_entry.model");
+const {
+  submitPlacementUpdateEntryInTimelines,
+} = require("../../utils/placement_entry_util");
+
+const addClientTimelineEntryRequestHandler = async (req, res) => {
+  try {
+    // eslint-disable-next-line camelcase
+    const { type, title, body, client, job_lead } = req.body.entry;
+
+    const { user } = req;
+
+    const validTypes = ["update", "contact", "placement", "note"];
+    if (!validTypes.includes(type)) {
+      return res.status(400).json({
+        status: "fail",
+        message: "Type not defined or is not valid",
+        data: null,
+      });
+    }
+
+    if (
+      ["update", "contact"].includes(type) &&
+      (!user || !title || !body || !client)
+    ) {
+      return res.status(400).json({
+        status: "fail",
+        message: `For type ${type}, user, title, body, and client must be defined`,
+        data: null,
+      });
+    }
+
+    if (
+      type === "placement" &&
+      // eslint-disable-next-line camelcase
+      (!user || !title || !body || !client || !job_lead)
```

Comment:
Is there a reason we are expecting a title and body on placement? I think these are being generated automatically by submitPlacementUpdateEntryInTimelines. This is also the case in the other timelines I believe

Reply:
I wasn't trying to make it so robust especially since this is only the backend api implementation, I was planning on polishing it and removing what I don't need in the frontend implementation to save a bit of time now so I can get started with the frontend ticket faster